ml-agents/mlagents/trainers/bc/trainer.py

```diff
@@ -112,26 +112,24 @@ def is_ready_update(self):
         Returns whether or not the trainer has enough elements to run update model
         :return: A boolean corresponding to whether or not update_model() can be run
         """
-        return (
-            len(self.demonstration_buffer.update_buffer["actions"]) > self.n_sequences
-        )
+        return len(self.demonstration_buffer["actions"]) > self.n_sequences
```

Comment:
Maybe the AgentBuffer should have a `get_len()` method, we seem to be calling `len(self.demonstration_buffer["actions"])` a lot.

Reply:
Or `get_num_demonstrations()`? Anything with `len` in it could be confusing, and overriding `__len__()` is likely to cause bugs.

Reply:
Agreed. In the AgentBuffer sample method I use `len(next(iter(self.values())))` to get the length of the first array. I'll wrap this into a public function

Reply:
Added a property called `num_experiences` in line with the future AgentExperiences